StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028CodeFixProvider.cs

```diff
@@ -71,27 +86,55 @@ public sealed override FixAllProvider GetFixAllProvider()
+                    newRoot = root.ReplaceTrivia(trivia, SyntaxTriviaList.Empty);
+                    break;
+                case SyntaxKind.SingleLineCommentTrivia:
+                    string newTriviaContent = trivia.ToFullString().Substring(0, trivia.FullSpan.Length - diagnosticSpan.Length);
+                    var newTrivia = SyntaxFactory.SyntaxTrivia(SyntaxKind.SingleLineCommentTrivia, newTriviaContent);
+                    newRoot = root.ReplaceTrivia(trivia, newTrivia);
+                    break;
+                case SyntaxKind.MultiLineCommentTrivia:
+                case SyntaxKind.MultiLineDocumentationCommentTrivia:
+                    var triviaLocation = trivia.GetLocation();
+                    string oldTriviaContent = trivia.ToFullString();
+                    TextSpan diagnosticSpanWithinTrivia = TextSpan.FromBounds(diagnosticSpan.Start - triviaLocation.SourceSpan.Start, diagnosticSpan.End - triviaLocation.SourceSpan.Start);
+                    newTriviaContent = string.Concat(
+                        oldTriviaContent.Substring(0, diagnosticSpanWithinTrivia.Start),
+                        oldTriviaContent.Substring(diagnosticSpanWithinTrivia.End));
+                    newTrivia = SyntaxFactory.SyntaxTrivia(trivia.Kind(), newTriviaContent);
+                    newRoot = root.ReplaceTrivia(trivia, newTrivia);
+                    break;
+                default:
+                    var node = root.FindNode(diagnosticSpan, findInsideTrivia: true, getInnermostNodeForTie: true);
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.XmlText:
+                            var xmlNode = (XmlTextSyntax)node;
+                            var newTextTokens = xmlNode.TextTokens;
+                            foreach (var textToken in newTextTokens)
+                            {
+                                if (textToken.Span.IntersectsWith(diagnosticSpan))
+                                {
+                                    diagnosticSpanWithinTrivia = TextSpan.FromBounds(diagnosticSpan.Start - textToken.Span.Start, diagnosticSpan.End - textToken.Span.Start);
+                                    oldTriviaContent = textToken.ValueText;
+                                    newTriviaContent = string.Concat(
+                                        oldTriviaContent.Substring(0, diagnosticSpanWithinTrivia.Start),
+                                        oldTriviaContent.Substring(diagnosticSpanWithinTrivia.End));
+                                    var newToken = SyntaxFactory.Token(textToken.LeadingTrivia, textToken.Kind(), newTriviaContent, newTriviaContent, textToken.TrailingTrivia);
+                                    newTextTokens = newTextTokens.Replace(textToken, newToken);
+                                    break;
+                                }
+                            }
+
+                            var newNode = xmlNode.Update(newTextTokens);
```

Comment:
And also: I couldn't find the `WithTextTokens` method for the life of me at the time I was coding this. I suspect I was trying before adding the downcast code I guess.